### PR TITLE
Change E0259 to the new error format

### LINKED ISSUE #35514: E0259 needs to be updated to new format
From: src/test/compile-fail/E0259.rs

E0259 needs a span_label, updating it from:

```
error[E0259]: an extern crate named `collections` has already been imported in this module
  --> src/test/compile-fail/E0259.rs:12:1
   |
11 | extern crate collections;
   | ------------------------- previous import of `collections` here
12 | extern crate libc as collections; //~ ERROR E0259
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0259]: an extern crate named `collections` has already been imported in this module
  --> src/test/compile-fail/E0259.rs:12:1
   |
11 | extern crate collections;
   | ------------------------- previous import of `collections` here
12 | extern crate libc as collections; //~ ERROR E0259
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ already imported
```

### PR DESCRIPTION
**Pull request by EugeneGonzalez**
Fixes #35514 as part of #35233.

Sorry about creating a new PR I was having a lot of troubles squashing the commit because I didn't properly branch the new feature.

r? @GuillaumeGomez

### CODE REVIEW

File: src/test/compile-fail/E0259.rs [hunk 1]
```diff
@@ -10,5 +10,6 @@
 
 extern crate collections;
 extern crate libc as collections; //~ ERROR E0259
+//~^ already imported
```
Comment (GuillaumeGomez) [inline src/test/compile-fail/E0259.rs]: Is this working? I have a little doubt.

Comment (jonathandturner) [inline src/test/compile-fail/E0259.rs]: @EugeneGonzalez - please change this label to be include NOTE as shown in the [blog post](http://www.jonathanturner.org/2016/08/helping-out-with-rust-errors.html)

**Comment by EugeneGonzalez** [comment]
@jonathandturner and @GuillaumeGomez sorry for the delay, I just got back to school. The Note is now added to both lines that have notes.

**Comment by EugeneGonzalez** [comment]
I noticed that #36100 implemented the bonus and that same approach could apply to the primary note of E0259. Should that also be added in?

**Comment by GuillaumeGomez** [comment]
It depends. If you feel that you can do it, then just do it and we'll review once updated. We're not in a hurry.

**Comment by EugeneGonzalez** [comment]
Seems this failed due to the same cause as #36138.

**Comment by jonathandturner** [comment]
@EugeneGonzalez - yeah, travis is having problems.

Btw, did you attempt the bonus by chance?

**Comment by EugeneGonzalez** [comment]
@jonathandturner there wasn't a bonus associated with this one, but I adapted the solution used in #36100 to get the form of
`'(conflicting crate name)' was already imported`
on the primary label.

**Comment by jonathandturner** [comment]
@EugeneGonzalez oh! okay.  Just let me know what it's ready for review and I'll take a look

**Comment by EugeneGonzalez** [comment]
@jonathandturner it is ready to be looked at. One concern I have is the different word used in the label. #36100 didn't use "was" while I did. It may be better to drop it for consistency, but I believe adding "was" makes a nicer label.

**Comment by jonathandturner** [comment]
@EugeneGonzalez - I like having the "was".  It does read a little better.  This looks good to go, so I'm going to go ahead and approve.

@bors r+ rollup

**Pull request merged**